medmodels/data_synthesis/mtgan/mtgan.pyi

```diff
@@ -0,0 +1,78 @@
+from pathlib import Path
+from typing import Dict, Optional
+
+from medmodels.data_synthesis.mtgan.builder import MTGANBuilder
+from medmodels.data_synthesis.mtgan.model.generator.generator import Generator
+from medmodels.data_synthesis.mtgan.model.real_gru.real_gru import RealGRU
+from medmodels.data_synthesis.mtgan.modules.postprocessor import (
+    AttributeType,
+    MTGANPostprocessor,
+    PostprocessingHyperparameters,
+    PostprocessingHyperparametersTotal,
+)
+from medmodels.data_synthesis.mtgan.modules.preprocessor import (
+    MTGANPreprocessor,
+    PreprocessingHyperparameters,
+    PreprocessingHyperparametersTotal,
+)
+from medmodels.data_synthesis.mtgan.mtgan_model import MTGANModel
+from medmodels.data_synthesis.mtgan.train.gan_trainer import (
+    TrainingHyperparameters,
+    TrainingHyperparametersTotal,
+)
+from medmodels.data_synthesis.synthesizer import Synthesizer
+from medmodels.medrecord.medrecord import MedRecord
+from medmodels.medrecord.types import MedRecordAttribute
+
+class MTGAN(Synthesizer):
+    generator: Optional[Generator]
```

Comment:
why does this need a generator? Doesn't the Generators' `GRU` hold information about a medrecord?

Reply:
you are right, it does not have to.